src/main/java/org/traccar/api/resource/PositionResource.java

```diff
@@ -86,6 +88,25 @@ public Collection<Position> getJson(
         }
     }
 
+    @DELETE
+    public Response remove(
+            @QueryParam("deviceId") long deviceId,
+            @QueryParam("from") Date from, @QueryParam("to") Date to) throws StorageException {
+        if (from == null || to == null || deviceId <= 0) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        permissionsService.checkPermission(Device.class, getUserId(), deviceId);
+        permissionsService.checkRestriction(getUserId(), UserRestrictions::getReadonly);
+        permissionsService.checkRestriction(getUserId(), UserRestrictions::getDeviceReadonly);
```

Comment:
Not sure about this one. I think "device readonly" user should still be able to remove positions.

Reply:
I haven't explored the source enough to know what "readonly" and "device readonly" actually mean. Will remove the check for "device readonly"